server/plugins/renderer/renderer.module.js

```diff
@@ -332,69 +334,30 @@ internals.redirect = function (response, request, callback) {
 };
 
 /**
- * Find and return the corresponding template file name when the needle is found
  *
- * @example
- * var haystack = {
- *   "products": {
- *     "dress.html": "/sample-marc-retro-style-summer-mid-dress/",
- *     "alligator.html": ["/sample-collette-alligator-clutch/", "/sample-jimmy-choo-extra-high-dynamite-cheetahs/"]
- *   },
- *   "search": {
- *     "search.html": "/example"
- *   }
- * };
- *
- * findDeepTemplate(haystack, "/example"); // -> "search.html"
- *
- * @param {Object} haystack
- * @param {String} needle
- * @return {String|Boolean}
- */
-internals.findDeepTemplate = function (haystack, needle) {
-    var template = false;
-
-    _.forEach(haystack, function (val, key) {
-        if (typeof val.length === 'undefined') {
-            // object, recursive find
-            template = internals.findDeepTemplate(val, needle);
-            if (template) {
-                template = (key + '/' + template).replace('.html', '');
-                return false; // break out of loop
-            }
-        } else if (typeof val === 'string') {
-            // found in string
-            if (val === needle) {
-                // found in string
-                template = key;
-                return false; // break out of loop
-            }
-        } else if (typeof val.length === 'number' && val.indexOf(needle) !== -1) {
-            // found in array
-            template = key;
-            return false; // break out of loop
-        }
-    });
-
-    return template;
-};
-
-/**
- *
- * @param {Object} request
- * @param {String} defaultTemplateFile
+ * @param {String} path
+ * @param {Object} data
  * @returns {string}
  */
-internals.getTemplatePath = function (request, defaultTemplateFile) {
-    var customLayouts = internals.options.customLayouts;
-    var templatePath = internals.findDeepTemplate(customLayouts, request.path);
+internals.getTemplatePath = function (path, data) {
+    var customLayouts = internals.options.customLayouts || {};
+    var pageType = data.page_type;
+    var templatePath;
+
+    if (internals.validCustomTemplatePageTypes.indexOf(pageType) && _.isPlainObject(customLayouts[pageType])) {
+        templatePath = _.findKey(customLayouts[pageType], function(p) {
```

Comment:
I haven't dig into this yet so I am not sure what `customLayouts` return, if `customLayouts[pageType]` contains more than object, are you aware that ``` `_.findkey` returns the key of the first element predicate returns truthy for```